python/ray/data/_internal/plan.py

```diff
@@ -164,7 +158,10 @@ def generate_logical_plan_string(
                     count = None
                 else:
                     assert len(sources) == 1
-                    plan = ExecutionPlan(DatasetStats(metadata={}, parent=None))
+                    plan = ExecutionPlan(
+                        DatasetStats(metadata={}, parent=None),
+                        self._context.copy(),
```

Comment:
Should we move copying inside the `ExecutionPlan` to make it less brittle?

Reply:
actually no need to copy here.